common/numerated/src/tree.rs

```diff
@@ -236,16 +234,16 @@ impl<T: Numerated> IntervalsTree<T> {
         let end = right_end.max(end);
 
         let Some((left_start, left_end)) = left_interval else {
+            debug_assert!(start <= end, "Must be cause of method it was found");
             self.inner.insert(start, end);
             return;
         };
 
         debug_assert!(left_end < right_start);
         debug_assert!(left_start <= start);
         let Some(left_end) = left_end.inc_if_lt(right_start) else {
-            unreachable!(
-                "`T: Numerated` impl error: for each x: T, y: T, x < y ↔ x.inc_if_lt(y) == Some(_)"
-            );
+            debug_assert!(false, "`T: Numerated` impl error");
+            return;
         };
 
         if left_end >= start {
```

Comment:
`self.inner.insert(left_start.max(start), end)`
maybe add specific private fn like `self.put(.., ..)` with internal debug assert and track caller

Reply:
nice, done